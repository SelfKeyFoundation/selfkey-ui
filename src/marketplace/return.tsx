import * as React from 'react';
import injectSheet, { StyleSheet } from 'react-jss';
import { Grid, WithStyles } from '@material-ui/core';
import { P } from '../typography/paragraph';
import ReturnIcon from '../icons/return-large';
import { TransactionFeeSelector } from '../transaction/transaction-fee-selector';
import CommonStyle from '../common/common-style';
import { StyledButton } from '../common/styled-button';

const styles: StyleSheet = {
	text: {
		fontSize: '18px',
		lineHeight: '30px',
	},
	footer: {
		marginTop: '30px',
		paddingTop: '30px',
		borderTop: '1px solid #475768',
	},
	contentSection: {
		marginBottom: '20px',
		marginTop: '20px',
	},
	understandLabel: {
		fontFamily: CommonStyle.fontFamily,
		fontSize: '14px',
		lineHeight: '21px',
		color: '#93B0C1',
	},
	understandCheckbox: {
		color: '#00C0D9',
		'&$primary$checked': {
			color: '#00C0D9',
		},
	},
	actions: {
		'&>button': {
			marginRight: '20px',
			marginTop: '30px',
		},
	},
	primary: {},
	checked: {},
};

export type ReturnState = {
	understood: boolean;
	error: boolean;
	fee: number;
};

export type ReturnProps = {
	amount?: number;
	token?: string;
	minGasPrice: number;
	maxGasPrice: number;
	gasLimit: number;
	fiat: string;
	fiatRate: number;
	onTransactionFeeChange?: (value: number) => void;
	onConfirm: (value: number) => void;
	onCancel: () => void;
};

export type StyledProps = WithStyles<keyof typeof styles> & ReturnProps;

export class ReturnComponent extends React.Component<StyledProps, ReturnState> {
	state = {
		error: false,
		understood: false,
		fee: 0,
	};
	handleFeeChange = (fee: number) => {
		this.setState({ fee });
	};
	handleConfirm = (event: any) => {
		event.preventDefault();
		return this.props.onConfirm(this.state.fee);
	};

	render() {
		const {
			classes,
			amount = 25,
			token = 'KEY',
			minGasPrice,
			maxGasPrice,
			gasLimit,
			fiat,
			fiatRate,
			onCancel,
		} = this.props;
		return (
			<Grid container direction="row" justify="flex-start" alignItems="flex-start">
				<Grid item xs={2}>
					<ReturnIcon />
				</Grid>
				<Grid item xs={10}>
					<Grid container direction="column" justify="flex-start" alignItems="stretch">
						<Grid item classes={{ item: classes.contentSection }}>
							<P className={classes.text}>
								You can now get your deposit of {amount} {token} tokens back. After the transaction is
								confirmed, you will lose access to this marketplace.
							</P>
						</Grid>
						<Grid item classes={{ item: classes.contentSection }}>
							<TransactionFeeSelector
								minGasPrice={minGasPrice}
								maxGasPrice={maxGasPrice}
								gasLimit={gasLimit}
								fiat={fiat}
								fiatRate={fiatRate}
								onChange={this.handleFeeChange}
							/>
						</Grid>
						<Grid item classes={{ item: classes.footer }}>
							<div className={classes.actions}>
								<StyledButton variant="contained" size="medium" onClick={this.handleConfirm}>
									Confirm
								</StyledButton>
								<StyledButton variant="outlined" size="medium" onClick={onCancel}>
									Cancel
								</StyledButton>
							</div>
						</Grid>
					</Grid>
				</Grid>
			</Grid>
		);
	}
}

export const Return = injectSheet(styles)<ReturnProps>(ReturnComponent);

export default Return;
