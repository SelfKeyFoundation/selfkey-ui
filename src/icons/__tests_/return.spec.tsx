import * as React from 'react';
import { create } from 'react-test-renderer';
import { ReturnIcon } from '../return';

describe('Return  Icon', () => {
	it('should show Return Icon', () => {
		const component = create(<ReturnIcon />);
		const tree = component.toJSON();
		expect(tree).toMatchSnapshot();
	});
});
