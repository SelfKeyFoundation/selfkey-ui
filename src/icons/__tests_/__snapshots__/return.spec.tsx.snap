// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Return  Icon should show Return Icon 1`] = `
<svg
  aria-hidden="true"
  className="MuiSvgIcon-root-1 Component-root-0-1-1"
  color={undefined}
  focusable="false"
  viewBox="0 0 15 18"
>
  <g
    fill="none"
    fill-rule="evenodd"
    id="ðŸ’¹-Marketplace"
    stroke="none"
    stroke-width="1"
  >
    <g
      fill="#1CA9BA"
      id="506.1---Exchange-Details-+-Active-Stake-+-Reclaim-KEY"
      transform="translate(-931.000000, -347.000000)"
    >
      <g
        id="Modal"
        transform="translate(240.000000, 146.000000)"
      >
        <g
          id="Area-Body"
          transform="translate(0.000000, 71.000000)"
        >
          <g
            id="Row-1"
            transform="translate(29.000000, 48.000000)"
          >
            <g
              id="Button/Full-Primary"
              transform="translate(638.000000, 69.000000)"
            >
              <g
                id="text"
                transform="translate(24.000000, 12.000000)"
              >
                <path
                  d="M7.30015817,14.1402847 L11.5546904,14.1402847 L9.18797907,11.7728434 L10.2202214,10.740601 L14.3499209,14.8703005 C12.8417082,16.3785132 11.6839032,17.5363183 10.2202214,19 L9.18797907,17.9677576 L11.5546904,15.6003163 L7.30015817,15.6003163 C3.26828081,15.6003163 0,12.3320355 0,8.30015817 C0,4.26828081 3.26828081,1 7.30015817,1 L14.6003163,1 L14.6003163,2.46003163 L7.30015817,2.46003163 C4.07494829,2.46003163 1.46003163,5.07494829 1.46003163,8.30015817 C1.46003163,11.525368 4.07494829,14.1402847 7.30015817,14.1402847"
                  id="icon-return"
                  transform="translate(7.300158, 10.000000) scale(-1, -1) translate(-7.300158, -10.000000) "
                />
              </g>
            </g>
          </g>
        </g>
      </g>
    </g>
  </g>
</svg>
`;
